src/receivers/HTTPReceiver.ts

```diff
@@ -302,19 +314,35 @@ export default class HTTPReceiver implements Receiver {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       const [installPath, installRedirectUriPath] = [this.installPath!, this.installRedirectUriPath!];
 
+      // Visiting the installation endpoint
       if (path === installPath) {
         // Render installation path (containing Add to Slack button)
         return this.handleInstallPathRequest(res);
       }
+
+      // Installation has been initiated
       if (path === installRedirectUriPath) {
         // Handle OAuth callback request (to exchange authorization grant for a new access token)
         return this.handleInstallRedirectRequest(req, res);
       }
     }
 
+    // Handle custom routes
+    if (this.customRoutes.length) {
+      const match = this.customRoutes.find((route) => {
```

Comment:
nice-to-have: If a developer has many custom routes and a custom route handles huge traffic, caching the traversal results here would be beneficial for slightly better performance. That being said, I think that this code won't cause visible bad effect in most cases.

Reply:
Great idea! Really appreciate you pushing for this. 

I took a second pass at it. What do you think of a cached object (processed right after the validation) that looks like the following:

```
{
  '/test': { 
     GET: [Function: handler], 
     POST: [Function: handler] 
   },
  '/health-check': {
     GET: [Function: handler],
     POST: [Function: handler],
     DELETE: [Function: handler]
   }
}
```

O(1) on retrieval and it also allows us to neatly match methods with their associated handlers, such as when a developer creates two separate routes with two separate handlers for the same `path`, as seen below (source for the output above):

```
customRoutes: [
    {
      path: '/test',
      method: ['GET', 'POST'],
      handler: (req, res) => {
        res.writeHead(200);
        res.end('Testing, testing.. 123!');
      },
    },
    {
      path: '/health-check',
      method: 'GET',
      handler: (req, res) => {
        res.writeHead(200);
        res.end('Health check information displayed here!');
      },
    },
    {
      path: '/health-check',
      method: ['POST', 'DELETE'],
      handler: (req, res) => {
        res.writeHead(200);
        res.end('Doing some other health check stuff!');
      },
    },
  ],
```

Reply:
No strong opinion on the details. Your approach should work.

Reply:
Looks great to me Alissa

Reply:
I've pushed up a refactor to incorporate the above. Review of the new additions requested and greatly appreciated!